astropy/coordinates/sky_coordinate.py

```diff
@@ -1685,9 +1685,16 @@ def from_name(cls, name, frame='icrs'):
             Instance of the SkyCoord class.
         """
 
+        from .jparser import jparser
         from .name_resolve import get_icrs_coordinates
 
-        icrs_coord = get_icrs_coordinates(name)
+        # first try extract coordinates from name. do this first since it
```

Comment:
I think one needs to be a bit careful here since the online catalog coordinates from sesame might be more accurate than the representation in the catalog name.

Reply:
Good point. This code was originally intended for objects names that were not yet included in the CDS database: eg. CRTS, MASTER etc. I see that those surveys are now aggregated by CDS, which means that this code is now less about extending the number of objects names that can be understood by `SkyCoord.from_name`, and more about potential performance gain.  As you say though, the actual catalog coordinates may be slightly different, so this comes at the cost of some accuracy - at most a fraction of a arcsec in the cases I've observed. I guess it would be more sensible to allow the J- name parsing as an option that the user can ask for if she doesn't care about sub-arcsec accuracy. Or to have it as a separate function entirely since it seems to muddy the waters a bit here.